spynnaker/pyNN/models/abstract_models/population_application_vertex.py

```diff
@@ -0,0 +1,328 @@
+        """
+        raise KeyError(
+            "This Population does not support the reading of state"
+            " variables")
+
+    def set_initial_state_values(self, name, value, selector=None):
+        """ Set the initial values of a state variable for the whole Population
+            or a subset if the selector is used
+
+        :param str name: The name of the variable to set
+        :param selector: a description of the subrange to accept, or ``None``
+            for all. See:
+            :py:meth:`~spinn_utilities.ranged.AbstractSized.selector_to_ids`
+        :type selector: None or slice or int or list(bool) or list(int)
+        :raise KeyError: if the variable is not something that can be changed
+        """
+        raise KeyError(
+            "This Population does not support the initialization of state"
+            " variables")
+
+    def get_current_state_values(self, names, selector=None):
+        """ Get the current values of a state variable for the whole Population
+            or a subset if the selector is used
+
+        :param names: The name or names of the variable to get
+        :type name: str or list
+        :param selector: a description of the subrange to accept, or ``None``
+            for all. See:
+            :py:meth:`~spinn_utilities.ranged.AbstractSized.selector_to_ids`
+        :type selector: None or slice or int or list(bool) or list(int)
+        :rtype: ParameterHolder
+        :raise KeyError: if the variable is not something that can be read
+        """
+        raise KeyError(
+            "This Population does not support the reading of state"
+            " variables")
+
+    def set_current_state_values(self, name, value, selector=None):
+        """ Set the current values of a state variable for the whole Population
+            or a subset if the selector is used
+
+        :param str name: The name of the variable to set
+        :param selector: a description of the subrange to accept, or ``None``
+            for all. See:
+            :py:meth:`~spinn_utilities.ranged.AbstractSized.selector_to_ids`
+        :type selector: None or slice or int or list(bool) or list(int)
+        :raise KeyError: if the variable is not something that can be changed
+        """
+        raise KeyError(
+            "This Population does not support the setting of state"
+            " variables")
+
+    def get_state_variables(self):
+        """ Get a list of supported state variables
+
+        :rtype: list(str)
+        """
+        return []
+
+    def get_units(self, name):
+        """ Get the units of the given parameter or state variable
+
+        :param str name: the name of the parameter to get the units of
+        :rtype: str
+        :raise KeyError:
+            if the name isn't recognised or the units cannot be identified
+        """
+        raise KeyError(f"The units for {name} cannot be found")
```

Comment:
I would say raise:
NotImplementedException("This Population does not support get_units")

As it is the whole call which went bad not the specific name

Reply:
As above, this allows the overridden call to call the super when it doesn't know what to do.